### PR TITLE
docs: Documentation on minValues in NodePool requirement.

### PR DESCRIPTION
**Pull request by nikmohan123**
**Description**
This PR adds details in the public doc related to `minValues` in the NodePool requirement.

RFC: https://github.com/kubernetes-sigs/karpenter/pull/966/files

**How was this change tested?**
`make presubmit`

**Does this change impact docs?**
- [x] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/v0.35/concepts/nodepools.md [hunk 1]
```diff
@@ -71,6 +71,11 @@ spec:
         - key: "karpenter.k8s.aws/instance-category"
           operator: In
           values: ["c", "m", "r"]
+          minValues: 2
+        - key: "karpenter.k8s.aws/instance-family"
+          operator: In
+          values: ["m5","m5d","c5","c5d","c4","r4"]
+          minValues: 5
```
Comment (jonathan-innis) [inline website/content/en/v0.35/concepts/nodepools.md:78]: Worth giving a short, high-level comment above this line about what this does? That way someone doesn't have to go down and read the section if they don't want to to understand it

File: website/content/en/v0.35/concepts/nodepools.md [hunk 2]
```diff
@@ -71,6 +71,11 @@ spec:
         - key: "karpenter.k8s.aws/instance-category"
```
[the remark this answers is not in the document]
Comment (nikmohan123) [reply website/content/en/v0.35/concepts/nodepools.md:71]: Got it. Thanks. Will do.

File: website/content/en/v0.35/concepts/nodepools.md (hunk 1 above)
Comment (nikmohan123) [reply website/content/en/v0.35/concepts/nodepools.md:81]: Addressed.

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**